diff --git a/bin.cjs b/bin.cjs
index 5e07847..f145feb 100755
--- a/bin.cjs
+++ b/bin.cjs
@@ -4650,7 +4650,7 @@ components:
       description: Token of the format "Bearer {token}" for normal authentication and "Convex {token}" for admin tokens.
   schemas:
 ${reindent(
-    functionSpec.functions.filter((f) => f.functionType !== "HttpAction").map((f) => generateEndpointSchemas(f)).join("\n"),
+    functionSpec.functions.filter((f) => f.functionType !== "HttpAction").filter((f) => includeInternal || f.visibility.kind === "public").map((f) => generateEndpointSchemas(f)).join("\n"),
     1
   )}
       FailedResponse:
